cocos/renderer/CCTextureCache.cpp

```diff
@@ -83,16 +83,43 @@ std::string TextureCache::getDescription() const
 struct TextureCache::AsyncStruct
 {
 public:
-    AsyncStruct(const std::string& fn, std::function<void(Texture2D*)> f) : filename(fn), callback(f), pixelFormat(Texture2D::getDefaultAlphaPixelFormat()), loadSuccess(false) {}
+    AsyncStruct(const std::string& fn, std::function<void(Texture2D*)> f) : filename(fn), callback(f), pixelFormat(Texture2D::getDefaultAlphaPixelFormat()), texture(nullptr), loadSuccess(false) {}
 
     std::string filename;
     std::function<void(Texture2D*)> callback;
     Image image;
     Image imageAlpha;
     Texture2D::PixelFormat pixelFormat;
+    Texture2D* texture; // optional, existing texture to init
```

Comment:
add a new field on TextureCache::AsyncStruct to indicate which texture should be inited